python/langsmith/evaluation/llm_evaluator.py

```diff
@@ -10,32 +11,66 @@
 
 
 class CategoricalScoreConfig(BaseModel):
-    """Configuration for a categorical score."""
+    """Configuration for a categorical score.
+
+    Attributes:
+        key (str): The feedback key for the evaluator.
+        choices (List[str]): List of valid categorical values that can be assigned.
+        description (str): Detailed description provided to the LLM judge of what
+            this score evaluates.
+        reasoning_key (Optional[str]): Key used to store the reasoning/explanation
+            for the score. Defaults to None.
+        reasoning_description (Optional[str]): Description provided to the LLM judge
+            of what should be included in the reasoning. Defaults to None.
+    """
 
     key: str
     choices: List[str]
     description: str
-    include_explanation: bool = False
-    explanation_description: Optional[str] = None
+    reasoning_key: Optional[str] = None
+    reasoning_description: Optional[str] = None
 
 
 class ContinuousScoreConfig(BaseModel):
-    """Configuration for a continuous score."""
+    """Configuration for a continuous numerical score.
+
+    Attributes:
+        key (str): The feedback key for the evaluator.
+        min (float): The minimum allowed value for the score. Defaults to 0.
+        max (float): The maximum allowed value for the score. Defaults to 1.
+        description (str): Detailed description provided to the LLM judge of what
+            this score evaluates.
+        reasoning_key (Optional[str]): Key used to store the reasoning/explanation
+            for the score. Defaults to None.
+        reasoning_description (Optional[str]): Description provided to the LLM judge
+            of what should be included in the reasoning. Defaults to None.
+    """
 
     key: str
     min: float = 0
     max: float = 1
     description: str
-    include_explanation: bool = False
-    explanation_description: Optional[str] = None
+    reasoning_key: Optional[str] = None
+    reasoning_description: Optional[str] = None
 
 
 def _create_score_json_schema(
     score_config: Union[CategoricalScoreConfig, ContinuousScoreConfig],
 ) -> dict:
     properties: Dict[str, Any] = {}
+
+    if score_config.reasoning_key:
+        properties[score_config.reasoning_key] = {
+            "type": "string",
+            "description": (
+                "The explanation for the score."
+                if score_config.reasoning_description is None
+                else score_config.reasoning_description
+            ),
+        }
+
     if isinstance(score_config, CategoricalScoreConfig):
-        properties["score"] = {
+        properties["value"] = {
             "type": "string",
             "enum": score_config.choices,
             "description": f"The score for the evaluation, one of "
```

Comment:
It isn't a score anymore, is it? It's the selected category.
The descriptions here matter a lot